quic/s2n-quic-core/src/havoc.rs

```diff
@@ -354,6 +391,38 @@ impl Strategy for Disabled {
     fn havoc<R: Random>(&mut self, _rand: &mut R, _buffer: &mut EncoderBuffer) {}
 }
 
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Hold<S: Strategy> {
+    strategy: S,
+    min: u64,
+    max: u64,
+    remaining: u64,
+}
+
+impl<S: Strategy> Hold<S> {
+    pub fn new(strategy: S, range: Range<usize>) -> Self {
+        debug_assert!(range.start <= range.end);
+        Self {
+            strategy,
+            min: range.start as _,
+            max: range.end as _,
+            remaining: 0,
+        }
+    }
+}
+
+impl<S: Strategy> Strategy for Hold<S> {
+    #[inline]
+    fn havoc<R: Random>(&mut self, rand: &mut R, buffer: &mut EncoderBuffer) {
+        if self.remaining == 0 {
+            self.strategy.havoc(rand, buffer);
+            self.remaining = rand.gen_range(self.min..self.max);
+        } else {
+            self.remaining -= 1;
```

Comment:
I probably should have been a bit more clear on the idea of holding a value... My thinking was you would copy the generated bytes in the encoder buffer out into a buffer on the strategy and replay those while `remaining > 0`. The way this is now it only havocs the value once per `remaining`.

### Current impl
```rust
input = 123;

[
    42,  // <- havoc
    123, // <- remaining = 2
    123, // <- remaining = 1
    123, // <- remaining = 0
    78,  // <- havoc
    123, // <- remaining = 2
    123, // <- remaining = 1
    123, // <- remaining = 0
]
```

### Ideal

```rust
input = 123;

[
    42,  // <- havoc
    42,  // <- remaining = 2
    42,  // <- remaining = 1
    42,  // <- remaining = 0
    78,  // <- havoc
    78,  // <- remaining = 2
    78,  // <- remaining = 1
    78,  // <- remaining = 0
]
```

Reply:
Ha, yeah you were clear enough, I was intending for it to have this behavior, just was a little confused on how some of the havoc code works

Reply:
No worries; it could probably use some documentation... But you can generally think of it as a pipeline of values, with each strategy potentially making some modification. The value produced by regular transport code is always provided as input in the EncoderBuffer, and then each strategy can decide what to do with that, passing it on to the next one. Since the transport code doesn't store the result anywhere, each of the strategies need to remember anything they need to perform their task.